compiler/modules/types/list.bal

```diff
@@ -270,15 +270,17 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, Conju
         // SemType d1 = diff(s[1], t[1]);
         // return !isEmpty(cx, d1) &&  tupleInhabited(cx, [s[0], d1], neg.rest);
         // We can generalize this to tuples of arbitrary length.
-        foreach int i in 0 ..< len {
-            SemType d = diff(fixedArrayGet(members, i), listMemberAt(nt.members, nt.rest, i));
-            if !isEmpty(cx, d) {
-                FixedLengthArray s = fixedArrayShallowCopy(members);
-                fixedArraySet(s, i, d);
-                if listInhabited(cx, s, rest, neg.next) {
+        foreach int i in 0 ..< members.initial.length() {
+            if listMemberInhabited(cx, neg, members, rest, nt, i) {
```

Comment:
Can we not combine these two loops? The first goes from 0 ..< members.initial.length() and the second goes from members.initial.length() ..< nt.members.initial.length()?

Why isn't it just a matter of changing `..< len` in the original code to `int:max(members.initial.length(), nt.members.initial.length())`

Reply:
Yes I missed that. Will merge the loops.

Reply:
If you combine the loops, you don't need to factor out the listMemberInhabited function (which means you don't need to think of a good name for it). Also it will be easier to see how the logic is changed.